lib/memo_wise.rb

```diff
@@ -94,42 +179,113 @@ def #{method_name}#{args_str}
     end
   end
 
-  def reset_memo_wise(method_name, *args, **kwargs)
+  # Presets the memoized result for the given method to the result of the given
+  # block.
+  #
+  # This method is for situations where the caller *already* has the result of
+  # an expensive method call, and wants to preset that result as memoized for
+  # future calls. In other words, the memoized method will be called *zero*
+  # times rather than once.
+  #
+  # @param method_name [Symbol]
+  #   Name of a method previously setup with `#memo_wise`.
+  #
+  # @param args [Array]
+  #   (Optional) If the method takes positional args, these are the values of
+  #   position args for which the given block's result will be preset as the
+  #   memoized result.
+  #
+  # @param kwargs [Hash]
+  #   (Optional) If the method takes keyword args, these are the keys and values
+  #   of keyword args for which the given block's result will be preset as the
+  #   memoized result.
+  #
+  # @yieldreturn [Object]
+  #   The result of the given block will be preset as memoized for future calls
+  #   to the given method.
+  #
+  # @return [void]
+  #
+  # @example
+  #   class Example
+  #     prepend MemoWise
+  #     attr_accessor :method_called_times
+  #
+  #     def method_to_preset
+  #       method_called_times = (method_called_times || 0) + 1
+  #       "A"
+  #     end
+  #     memo_wise :method_to_preset
+  #   end
+  #
+  #   ex = Example.new
+  #
+  #   ex.preset_memo_wise(:method_to_preset) { "B" }
+  #
+  #   ex.method_to_preset #=> "B"
+  #
+  #   ex.method_called_times #=> nil
+  #
+  def preset_memo_wise(method_name, *args, **kwargs)
     validate_memo_wised!(method_name)
 
-    unless method_name.is_a?(Symbol)
-      raise ArgumentError, "#{method_name.inspect} must be a Symbol"
+    unless block_given?
+      raise ArgumentError,
+            "Pass a block as the value to preset for #{method_name}, #{args}"
     end
 
-    unless respond_to?(method_name)
-      raise ArgumentError, "#{method_name} is not a defined method"
-    end
+    validate_params!(method_name, args)
 
-    if args.empty? && kwargs.empty?
-      @_memo_wise.delete(method_name)
+    if method(method_name).arity.zero?
+      @_memo_wise[method_name] = yield
     else
-      @_memo_wise[method_name].delete(fetch_key(method_name, *args, **kwargs))
+      @_memo_wise[method_name][fetch_key(method_name, *args, **kwargs)] = yield
     end
   end
 
+  # Resets all memoized results -- of all methods.
+  #
+  # @return [void]
```

Comment:
Agreed: remove this, and make `method_name` optional on `reset_memo_wise` for this